mono/metadata/handle.h

```diff
@@ -341,12 +341,15 @@ void mono_handle_verify (MonoRawHandle handle);
 #define MONO_HANDLE_PAYLOAD_OFFSET_(PayloadType) MONO_STRUCT_OFFSET(PayloadType, __raw)
 #define MONO_HANDLE_PAYLOAD_OFFSET(TYPE) MONO_HANDLE_PAYLOAD_OFFSET_(TYPED_HANDLE_PAYLOAD_NAME (TYPE))
 
-#define MONO_HANDLE_INIT ((void*) mono_null_value_handle)
 #define NULL_HANDLE mono_null_value_handle
 
 //XXX add functions to get/set raw, set field, set field to null, set array, set array to null
 #define MONO_HANDLE_RAW(HANDLE) (HANDLE_INVARIANTS (HANDLE), ((HANDLE)->__raw))
 #define MONO_HANDLE_DCL(TYPE, NAME) TYPED_HANDLE_NAME(TYPE) NAME = MONO_HANDLE_NEW (TYPE, (NAME ## _raw))
+#define MONO_HANDLE_LOCAL_VARIABLE_NOT_INITIALIZED(type, name) TYPED_HANDLE_NAME(type) name
+#define MONO_HANDLE_LOCAL_VARIABLE_INITIALIZED_NULL(type, name) \
+	MONO_HANDLE_LOCAL_VARIABLE_NOT_INITIALIZED (type, name) = MONO_HANDLE_NEW (type, NULL)
+#define MONO_HANDLE_SET_NULL(type, name) ((name) = MONO_HANDLE_NEW (type, NULL))
```

Comment:
I don't think `MONO_HANDLE_SET_NULL` has the semantics that we want.

Suppose I have `MonoObjectHandle p;` in scope and it was previously initialized to point to some `MonoObject *o1`.   What `MONO_HANDLE_SET_NULL (MonoObject, p)` will do is re-assign `p` to a *new* handle that's initialized to null.  The old handle is silently discarded (although it's still allocated in its handle frame and still holds a pointer to `o1`.

What I think we want instead is

```c
#define MONO_HANDLE_ASSIGN_NULL(h) (MONO_HANDLE_SUPPRESS ((MONO_HANDLE_UNSUPPRESS (h))->__raw = NULL)
/* essentially h->__raw = NULL */
```

In the example, `MONO_HANDLE_ASSIGN_NULL (p)`,  that will set the existing handle `p` to point to `NULL` instead of to `o1` without allocating any more handles.

Reply:
ok I think I understand better now, thank you, I'll remove this new macro, at least